mkosi/distributions/gentoo.py

```diff
@@ -82,117 +118,40 @@ class Gentoo:
-            logging.info(PORTAGE_INSTALL_INSTRUCTIONS)
-            raise e
-
-        return dict(profile_path=PROFILE_PATH,
-                    custom_profile_path=CUSTOM_PROFILE_PATH,
-                    ebuild_sh_env_dir=EBUILD_SH_ENV_DIR,
-                    portage_cfg_dir=USER_CONFIG_PATH)
-
-    @complete_step("Installing Gentoo…")
-    def __init__(self, state: MkosiState) -> None:
-        # TOCLEANUP: legacy namig, to be cleaned up
-        self.state = state
-        self.config = self.state.config
-        self.root = self.state.root
-        ret = self.try_import_portage()
-
-        from portage.package.ebuild.config import config as portage_cfg  # type: ignore
-
-        self.portage_cfg = portage_cfg(config_root=str(state.root),
-                                       target_root=str(state.root),
-                                       sysroot=str(state.root), eprefix=None)
-
-        PORTAGE_MISCONFIGURED_MSG = "Missing defaults for portage, bailing out"
-        # we check for PORTDIR, but we could check for any other one
-        if self.portage_cfg['PORTDIR'] is None:
-            die(PORTAGE_MISCONFIGURED_MSG)
-
-        self.profile_path = state.root / ret["profile_path"]
-        self.custom_profile_path = state.root / ret["custom_profile_path"]
-        self.ebuild_sh_env_dir = state.root / ret["ebuild_sh_env_dir"]
-        self.portage_cfg_dir = state.root / ret["portage_cfg_dir"]
-
-        self.portage_cfg_dir.mkdir(parents=True, exist_ok=True)
-
-        self.arch, _ = state.installer.architecture(state.config.architecture)
-        self.arch_profile = Path(f"default/linux/{self.arch}/{state.config.release}/no-multilib/systemd/merged-usr")
-        self.pkgs['sys'] = ["@world"]
-
-        self.pkgs['boot'] = [
-            "sys-kernel/installkernel-systemd-boot",
-            "sys-kernel/gentoo-kernel-bin",
-        ]
+    def kernel_image(name: str, architecture: Architecture) -> Path:
+        kimg_path = {
+            Architecture.x86_64: "arch/x86/boot/bzImage",
+            Architecture.arm64: "arch/arm64/boot/Image.gz",
+            Architecture.arm: "arch/arm/boot/zImage",
+        }[architecture]
 
-        self.emerge_vars = {
-            "FEATURES": " ".join(self.portage_features),
-            "USE": " ".join(self.portage_use_flags),
-        }
+        return Path(f"usr/src/linux-{name}") / kimg_path
 
-        self.fetch_fix_stage3()
-        self.set_useflags()
-        self.mkosi_conf()
-        self.get_snapshot_of_portage_tree()
-        self.update_stage3()
-        self.depclean()
+    @classmethod
+    def install(cls, state: MkosiState) -> None:
+        config = state.config
+        root = state.root
+
+        arch = state.installer.architecture(state.config.architecture)
 
-    def fetch_fix_stage3(self) -> None:
         """usrmerge tracker bug: https://bugs.gentoo.org/690294"""
 
+        if config.mirror:
+            cls.gentoo_mirrors = config.mirror
```

Comment:
Let's drop this and just use `config.mirror` directly.

Reply:
what if `config.mirror` is `None`? `MkosiConfig` is `frozen`.

Reply:
You can set a default mirror in config_default_mirror() in config.py

Reply:
done!